packages/notifi-frontend-client/lib/client/ensureSource.ts

```diff
@@ -474,6 +474,9 @@ const ensureSources = async (
     case 'label': {
       throw new Error('Unsupported event type');
     }
+    case 'fusionMultiThreshhold': {
+      throw new Error('Unsupported event type');
```

Comment:
Oh.. I think there is a typo, maybe `fusionMultiThreshold`?

And because we can reuse the `ensureFusionSource` here, we can just put this case above `case 'fusion'` as below:
```
// ....
    case 'fusionMultiThreshold'
    case 'fusionToggle':
    case 'fusion': {
      const source = await ensureFusionSource(service, eventType, inputs);
      return [source];
    }
```

Reply:
`fusionMultiThreshold` is not really an alert itself, it is just a wrapper for multiple `fusion` alerts, so I don't think we will ever want to ensureAlert with an actual fusionMultiThreshold eventtypeitem

Reply:
Oh yeah ... it is similar as `TradingPair` that is actually a `DirectPush` source with it's own `filterOption`.
So we might be able to reuse the existing `ensureSource` for fusion alert. 
CC: @juni-notifi: Hope I understand it correctly.

Reply:
yeah the way i have designed it is that I'm just subscribing a `fusion` source, and never a `fusionMultiThreshold` source, similar to `TradingPair`. In this case every alert is the same as the fusion health check alert so I'm reusing all the fusion filters and sources, thus I have 

`case 'fusionMultiThreshhold': {
      throw new Error('Unsupported event type');
    }`

and also in the subscribeAlert you can see I'm just creating and subscribing a fusion alert https://github.com/notifi-network/notifi-sdk-ts/pull/382/files#diff-ec97d4ca3cd1d1913fc81a7184a0520b17c17d4c913688eadc505b43231702e7R315

Reply:
Oh ... I see, we convert `FusionMultiThresholdEventType` to `FusionEventTypeItem`  (HEALTH_CHECK). Got it.

That case, do we have to define a new type (`FusionMultiThresholdEventType`). or can we reuse the fusion health type simply.

Reply:
I think we need the `FusionMultiThresholdEventType` to render the correct component in the card, otherwise it would render the fusion health check row component

Reply:
The last question is that why cannot we reuse the `FusionEventTypeItem` with `FusionTypeBase`.
How do you think using `FusionEventTypeItem` with `FusionTypeBase` and define the `selectedUIType=MULTI_THRESHHOLD`. Then instead of throwing error in `ensureSourceAndFilters`, we can just simply reuse `ensureFusionSource`.
And just need to change `getFusionSouceFilter` as below.

````ts
const getFusionSourceFilter = (
  //...
  else if (eventType.selectedUIType === 'HEALTH_CHECK' || eventType.selectedUIType === 'MULTI_THRESHHOLD') {
    // ....
  }
```

Reply:
I dont think we would ever want to `ensureSourceAndFilters` with a multithreshold eventtype, since we only want one (so only one row renders in the card) so I would definitely keep throwing the error in `ensureSourceAndFilters`. 

However I think we can reuse `FusionTypeBase` at the type declaration to reduce redundancy and use `selectedUIType` to make sure the right component renders, let me look into that

Reply:
added a change to use `FusionTypeBase`, for the getFusionSourceFilter I don't think we want to ever call ensureAlert with a fusionMulti event type so I would not change it